### ISSUE TITLE
Chunking up chi^2

### ISSUE THREAD

**Issue by bennahugo** [ratt-ru/CubiCal#158]
Trying to compute bandpasses for MK 46 and running out of memory on smallest possible interval

Back of the envelope calculations show a 10 minute interval (ensuring -30dB level solution stability) will be 79 GiB of data. Presumably a model data column of the same size is created as well.  I've already turned off course grained parallelism.
```
gocubical --out-column DATA --data-ms msdir/selfcal.ms --data-column CORRECTED_DATA --model-list output/0407-65.fitted.txt --model-ddes never --weight-column WEIGHT --flags-auto-init 0 --flags-save 0 --sol-jones G --data-freq-chunk 0 --data-time-chunk 300 --g-
time-int 300 --g-freq-int 1 --g-type complex-2x2 --sol-min-bl 8.0 --sel-field 0 --dist-ncpu 1 --dist-min-chunks 1 --g-max-iter 60
```

Memory usage hovers between 450 and 500 GiB. Presumably the X^2 can be computed in much smaller chunks and the data reread from disk to apply - similar to a CASA workflow. I can see this becoming intractable with MK64

**Comment by IanHeywood** [comment]
I guess there are 4k channels here. I think @o-smirnov is working on an on-the-fly averaging fix, but until then a workaround for bandpass cal might be to not have --date-freq-chunk set to zero but to solve in sub-bands and concatenate the tables afterwards.

**Comment by bennahugo** [comment]
Yup.. I've resorted to that now, unfortunately it means it is striding through disk...

**Comment by IanHeywood** [comment]
Think there's any mileage in using TaQL to select half the baselines, then solving for the other half and averaging the pair of tables that result?

**Comment by bennahugo** [comment]
Possibly unless we're working with a subset that is dominated by residual RFI - I still think it will be quicker to run through the data twice sequentially. I think this is how CASA keeps its memory footprint within bounds

**Comment by o-smirnov** [comment]
#151 is intended to address this... gimme a day or two...
